### DISCUSSION TITLE
valueAsDate does not behave as per the documentation

### DISCUSSION THREAD

**Discussion by mrmanicou** [react-hook-form/react-hook-form#5879] [General] [answered]
According to the documentation:

> valueAsDate: Returns a Date normally. If something goes wrong null will be returned.

 But null never gets returned. A peek into the code shows this line:

`const valueDate =
        (ref as HTMLInputElement).valueAsDate || new Date(inputValue);`

What this implies is that RHF will first try to use the native value as date function which indeed returns `null` if invalid Date is encountered but the `new Date(inputValue)` part 

> with an invalid date string, it returns a Date object whose toString() method returns the literal string Invalid Date.

This according to the Javascript docs.

I like the behaviour given in the docs, but the code will never give these results. Instead I'll to use `setValueAs` and do my own thing? Is this what others do or am I missing something?

**Comment by bluebill1049** [comment, marked as answer]
I think i will update the doc on this, return null is probably not appropriate for input, since input will always default to `string`
